File: app/src/main/java/com/chooloo/www/callmanager/ui/base/MvpView.java
package com.chooloo.www.callmanager.ui.base;

import androidx.annotation.StringRes;

public interface MvpView {

    String[] onGetPermissions();

    void onSetup();

    boolean hasPermissions();

    boolean hasPermission(String permission);

    void askForPermission(String permission, int requestCode);

    void askForPermissions(String[] permissions, int requestCode);

    void askForPermissions();

    void showMessage(String message);

    void showMessage(@StringRes int stringResId);

    void showError(String message);

    void showError(@StringRes int stringResId);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/base/MvpView.kt
package com.chooloo.www.callmanager.ui.base

import androidx.annotation.StringRes

interface MvpView {
    fun onSetup()

    fun hasPermission(permission: String): Boolean
    fun hasPermissions(permissions: Array<String>): Boolean

    fun askForPermission(permission: String, requestCode: Int?)
    fun askForPermissions(permissions: Array<String>, requestCode: Int?)

    fun showMessage(message: String)
    fun showMessage(@StringRes stringResId: Int)
    fun showError(message: String)
    fun showError(@StringRes stringResId: Int)
}